crates/node/src/types.rs

```diff
@@ -78,17 +61,20 @@ pub enum CatchupError {
     InternalError,
 }
 
-#[derive(Debug, Deserialize, Serialize)]
+#[derive(Clone, Debug, Deserialize, Serialize)]
 #[non_exhaustive]
-pub struct TransactionWithStatus {
```

Comment:
Okay so my current understanding is that this should be removed - that seems to have been confirmed as correct 🙂

Reply:
Sandi's concern is valid and we've since discussed this on a call.

For posterity, while we should remove the concept of "transactions", clients should maintain the same model for requesting execution (method name like `set` for kv-store or `create_post` for only-peers, for example), during which the "actions" should be generated by the storage subsystem following storage interactions during the execution of the client request.

Reply:
That's a mix of different things.

We don't have transactions anymore, which is why they have been removed. They have been replaced by the operations needed for CRDTs. This has been confirmed as correct.

Separately, the method names in examples, i.e. things used by clients, are arbitrary from my perspective and I don't see any issue maintaining those. If that mechanism is required to be a certain way, that's fine. However, that's with you at present I believe.

Yesterday you said that the `Action`s should be created via this `set()` method - that sounds fine to me; we appeared to be on the same page about how and where that should happen. However, if that needs to be moved or refined, I have no objection - feel free! 🙂